src/graphics/Screen.cpp

```diff
@@ -281,12 +281,27 @@ static void drawFunctionOverlay(OLEDDisplay *display, OLEDDisplayUiState *state)
 /// Used on eink displays while in deep sleep
 static void drawDeepSleepScreen(OLEDDisplay *display, OLEDDisplayUiState *state, int16_t x, int16_t y)
 {
+    const char *idText = owner.short_name;
+
     // Next frame should use full-refresh, and block while running, else device will sleep before async callback
     EINK_ADD_FRAMEFLAG(display, COSMETIC);
     EINK_ADD_FRAMEFLAG(display, BLOCKING);
 
     LOG_DEBUG("Drawing deep sleep screen\n");
-    drawIconScreen("Sleeping...", display, state, x, y);
+    
+    // Check if idText is empty
+    if (strlen(idText) == 0) {
+        drawIconScreen("Sleeping.", display, state, x, y);
+    } else {
+        // Create a new string that is the concatenation of idText and " sleeping."
+        size_t idTextLen = strlen(idText);
+        char displayStr[idTextLen + 12]; // Allocate enough space for the new string
+        strcpy(displayStr, idText); // Copy idText to the new string
```

Comment:
Perhaps more consolidated to write in this way with a single expression?
sprintf(displayStr, "%s sleeping.",  owner.short_name);

Reply:
Also, no need to declare idText this way. You can just reference owner.short_name

Reply:
Absolutely guys. I'm not a C++ programmer and just tried that the code works. I also don't like the strlen check on a maybe unassigned var. So if you guys know it better - pls change.

Reply:
Thought about something like 
if (!idText || strlen(idText) == 0) {
for the check of idText ?

Reply:
Actually, is it possible to get into a situation where short name is has zero length? It might not be an issue? Now I think about it though, I'm not sure what would happen with emoji short names. I should check if this is a problem with the existing E-Ink screensaver code too.